### ISSUE TITLE
Inconsistent handling of steps when using accumulate_grad_batches

### ISSUE THREAD

**Issue by bradezard131** [PyTorchLightning/pytorch-lightning#2817]
## 🐛 Bug

When using `accumulate_grad_batches` the steps are handled differently in the first epoch from any epoch later. Steps in the first epoch are counted for every batch regardless of accumulate_grad_batches, steps in later epochs are counted as one batch per accumulation (i.e. batch / accumulate_grad_batches). This makes learning rate scheduling, max iters, and logging very confusing

### To Reproduce

Train any model using `accumulate_grad_batches=2, check_val_every_n_epoch=1` and you'll see the iteration of the first eval is twice the difference between first and second.

### Expected behavior

It should be one or the other, and documented which it is. Doesn't really matter as long as it is consistent

**Comment by rohitgr7** [comment]
can you share a minimal example(colab) to check this issue?

**Comment by edenlightning** [comment]
Closing this for now, feel free to reopen if you can reproduce using [our base model example](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/pl_examples/bug_report_model.py)
